wallet/src/de/schildbach/wallet/data/TransactionMetadataDocument.kt

```diff
@@ -0,0 +1,20 @@
+package de.schildbach.wallet.data
+
+import androidx.room.Entity
+import org.bitcoinj.core.Sha256Hash
+import org.dash.wallet.common.data.TaxCategory
+
+@Entity(tableName = "transaction_metadata_platform", primaryKeys = ["id", "txId"])
+class TransactionMetadataDocument(
+    val id: String,
+    val timestamp: Long,
+    var txId: Sha256Hash,
+    var sentTimestamp: Long? = null,
+    var taxCategory: TaxCategory? = null,
+    var currencyCode: String? = null,
+    var rate: Double? = null,
+    var memo: String? = null,
+    var service: String? = null
+) {
+
+}
```

Comment:
On platform, `TransactionMetadataDocument` contain an encrypted list of transaction metadata.  Here the metadata will be stored with one transaction change per Platform document `id` per record.